### ISSUE TITLE
Chooser and SteamField panels currently do not use a "heading" argument

### ISSUE THREAD

**Issue by tbrlpld** [wagtail/wagtail#7849]
### Is your proposal related to a problem?

The chooser and SteamField panels currently do not allow to override the "heading" of the panel that is displayed in the admin.

`FieldPanel` and `MultiFieldPanel` do allow this. 

Enabling the override provides a nice way to hide often long and cryptic internal field names from the admin users and allows to set more friendly panel names. 

### Describe the solution you'd like

Allow the chooser and StreamField panels to accept a `heading` key word argument (just like `FieldPanel` and `MultiFieldPanel`).

If provided, the value of the "heading" should be used in stead of the label that is automatically generated from the field name.

### Describe alternatives you've considered

There is not much else possible.

**Comment by lb-** [comment]
To be reviewed, 3.0 brings more generic FieldPanel usage and this may be resolved.

**Comment by lb-** [comment]
This has been resolved in Wagtail 3.0 with the adoption of `FieldPanel` for all scenarios.

See https://docs.wagtail.org/en/latest/releases/3.0.html#removal-of-special-purpose-field-panel-types
See `FieldPanel` accepts optional `header` https://docs.wagtail.org/en/latest/reference/pages/panels.html#wagtail.admin.panels.FieldPanel

### Example
```
    content_panels = Page.content_panels + [
        FieldPanel('introduction', classname="full"),
        FieldPanel('image', heading='IMAGE CHOOSER (IMAGE) heading'),
        FieldPanel('body', heading='STREAMFIELD (BODY) HEADING'),
        FieldPanel('origin', heading='FIELDPANEL (ORIGIN) HEADING'),
        FieldPanel('bread_type'),
```

<img width="1646" alt="Screen Shot 2022-05-06 at 7 53 30 am" src="https://user-images.githubusercontent.com/1396140/167031908-5343758f-d7e4-4484-8c56-3bbc215cd5c5.png">
